helper/impls/helper-impl-110/src/main/java/com/linkedin/avroutil1/compatibility/avro110/Avro110AvscWriter.java

```diff
@@ -85,4 +96,36 @@ private void writeProps(Map<String, Object> props, Jackson2JsonGeneratorWrapper
             delegate.writeObjectField(entry.getKey(), JacksonUtils.toJsonNode(o));
         }
     }
+
+    /**
+     *  Enforces uniform numeric default values across Avro versions
+     */
+    private JsonNode enforceUniformNumericDefaultValues(Schema.Field field) {
+        JsonNode genericDefaultValue = Accessor.defaultValue(field);
+        if (!genericDefaultValue.isNumber()) {
+            LOGGER.warn(String.format("Invalid default value: %s for \"long\" field: %s", genericDefaultValue, field.name()));
+            return genericDefaultValue;
+        }
+        double numericDefaultValue = genericDefaultValue.doubleValue();
+        switch (field.schema().getType()) {
```

Comment:
fields can be unions?

Reply:
should be fixed in the most recent commit